juju1/provider/openstack/provider.go

```diff
@@ -1813,15 +1827,34 @@ func (e *environ) DowngradeTags() error {
 	return errors.Trace(e.downgradeGroups(modelUUID))
 }
 
+const securityGroupPrefix = `^juju-[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}-`
+
 func (e *environ) downgradeGroups(modelUUID string) error {
 	client := e.nova()
+	groupPrefixRe, err := regexp.Compile(securityGroupPrefix + modelUUID)
```

Comment:
It seems weird that you are adding a modelUUID here. You have two UUIDs to match?

Reply:
In this case I want to match any group that has the right model UUID, but I don't know the controller UUID (so match it with a regex). It's possible in the abort code higher up that we couldn't connect to the controller to find it out, but since the abort command is trying to undo as much as possible regardless of errors in an unrelated task, we still want to clean up these groups.